### PR TITLE
Add automaton parser

### PR DESCRIPTION
**Pull request by yihellen**
Add automaton parser to Gramatron so that AFLplusplus can run both Gramatron and Nautilus as custom mutators

### CODE REVIEW

**Comment by domenukk** [comment]
Thanks! Please direct PRs towards the dev branch in the future :)

**Comment by vanhauser-thc** [comment]
@yihellen thank you for this!
I see quite some changes, but what exactly is this PR doing?
And there is no update of the README.

File: custom_mutators/gramatron/build_gramatron_mutator.sh [hunk 1]
```diff
@@ -144,6 +144,6 @@ echo
 echo
 echo "[+] Json-c successfully prepared!"
 echo "[+] Builing gramatron now."
-$CC -O3 -g -fPIC -Wno-unused-result -Wl,--allow-multiple-definition -I../../include -o gramatron.so -shared -I. -I/prg/dev/include gramfuzz.c gramfuzz-helpers.c gramfuzz-mutators.c gramfuzz-util.c hashmap.c ../../src/afl-performance.o json-c/.libs/libjson-c.a || exit 1
+$CC -O0 -g -fPIC -Wno-unused-result -Wl,--allow-multiple-definition -I../../include -o gramatron.so -shared -I. -I/prg/dev/include gramfuzz.c gramfuzz-helpers.c gramfuzz-mutators.c gramfuzz-util.c hashmap.c automaton-parser.c ../../src/afl-performance.o json-c/.libs/libjson-c.a || exit 1
```
Comment (vanhauser-thc) [inline custom_mutators/gramatron/build_gramatron_mutator.sh:147]: for debug purposes O0 is fine, but I guess for the PR O2 or O3 would be better?
Or support CFLAGS, and set them to -O2 if not present?

File: custom_mutators/gramatron/testMakefile.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
+CC=gcc
```
Comment (vanhauser-thc) [inline custom_mutators/gramatron/testMakefile.mk:1]: maybe `?=` instead?

**Comment by yihellen** [comment]
Hi! Sorry for not seeing this sooner.
I must have messed with my git, I intended to make a push to my forked repo yesterday!
Thank you for reviewing my PR though!
I am trying to make Gramatron more compatible to run with other custom mutators by adding a automaton parser, but I haven't quite cleaned up my code yet. Sorry for the mess!

**Comment by vanhauser-thc** [comment]
@yihellen if you have something ready that works we can support in testing the effectiveness (we are the most active fuzzbench users by far ;)

**Pull request closed without merging**